docs/repositories.md

```diff
@@ -122,10 +122,19 @@ name = "foo"
 url = "https://foo.bar/simple/"
 default = false
 secondary = false
+explicit = false
 ```
 
-Any package source not marked as `secondary` will take precedence over [PyPI](https://pypi.org).
+If all keys are `false`, the source is considered a "primary" source that takes precedence over default, secondary and explicit sources.
 
+Package sources are considered in the following order:
+1. "primary" sources,
+2. [default source](#default-package-source),
+3. [secondary sources](#secondary-package-sources),
```

Comment:
Not quite correct. We shouldn't care about `default` too much here because I don't want to change default handling in this PR.

Reply:
I think I have made the MR compatible with the current behavior (as described in https://github.com/python-poetry/poetry/pull/6879#pullrequestreview-1273887207).
- `Priority.DEFAULT` is the highest priority.
- PyPI is disabled if another source has priority `DEFAULT`.
- PyPI has priority `DEFAULT` if no other sources are `DEFAULT` _or_ `PRIMARY`.
- PyPI has priority 'SECONDARY' if no other sources are `DEFAULT`, but at least one is `PRIMARY`.

As we now have several types of source priorities, it feels right to still mention the default and PyPI in the documentation. I kept it very minimal for now in line with your comment.

Please carefully review the latest commits.